tests/unit-tests/test-class-course.php

```diff
@@ -306,7 +306,37 @@ public function testLogNoEventProductId() {
 
 		// Ensure product ID is correct.
 		$event = $events[0];
-		$this->assertEquals( -1, $event['url_args']['product_id'] );
+		$this->assertEquals( 0, $event['url_args']['product_count'] );
 	}
 
+	/**
+	 * Test initial publish logging product count with multiple product IDs.
+	 *
+	 * @covers Sensei_Course::log_initial_publish_event
+	 */
+	public function testLogEventProductCountMultiProduct() {
+		$course_id = $this->factory->course->create(
+			[
+				'post_status' => 'draft',
+			]
+		);
+		add_post_meta( $course_id, '_course_woocommerce_product', 5 );
+		add_post_meta( $course_id, '_course_woocommerce_product', 6 );
+
+		// Publish.
+		wp_update_post(
+			[
+				'ID'          => $course_id,
+				'post_status' => 'publish',
+			]
+		);
+
+		Sensei()->post_types->fire_scheduled_initial_publish_actions();
+		$events = Sensei_Test_Events::get_logged_events( 'sensei_course_publish' );
+		$this->assertCount( 1, $events );
```

Comment:
If we're asserting multiple things in the same test case, it may be hard to tell which assertion is the one that is failing. Adding a third parameter to both of the assertions will facilitate debugging.